web-ng/src/app/components/form-field-editor/form-field-editor.component.ts

```diff
@@ -320,4 +321,14 @@ export class FormFieldEditorComponent implements OnInit, OnChanges, OnDestroy {
       editor.optionGroup.markAllAsTouched();
     });
   }
+
+  private focusNewOption(index: number): void {
+    // setTimeout is to added to wait for the option to load.
```

Comment:
How does this work? I would think this would result in a race condition, but may just be my own ignorance.

Reply:
If I am understanding your code correctly @Parul#0929, the reason you added the `setTimeout` is that if you don't set it, then it will execute too soon, so that it will focus on the second last item because angular thought that that was the last.  So you could force angular to re-render it before doing the focus. Here are some example how you can do it. https://stackoverflow.com/questions/35105374/how-to-force-a-components-re-rendering-in-angular-2 & https://stackoverflow.com/questions/43457149/angular-4-what-is-the-right-way-to-wait-for-operation.